lib/msf/core/auxiliary/vyos.rb

```diff
@@ -0,0 +1,302 @@
+        # jtr_format: 'sha512,crypt', # default on the devices 11.4.0+
+        service_name: '',
+        module_fullname: fullname,
+        status: Metasploit::Model::Login::Status::UNTRIED
+      }
+
+      # Default SNMP to UDP
+      if tport == 161
+        credential_data[:protocol] = 'udp'
+      end
+
+      if store && !config.include?('such file or directory')
+        l = store_loot('vyos.config', 'text/plain', thost, config.strip, 'config.txt', 'VyOS Configuration')
+        vprint_good("Config saved to: #{l}")
+      end
+
+      host_info = {
+        host: thost,
+        os_name: 'VyOS'
+      }
+      report_host(host_info)
+
+      # generated by: cat /config/config.boot
+      # https://github.com/rapid7/metasploit-framework/issues/14124
+
+      # login {
+      #    user jsmith {
+      #        authentication {
+      #            encrypted-password $6$ELBrDuW7c/8$nN7MwUST8s8O0R6HMNu/iPoTQ1s..y8HTnXraJ7Hh4bHefRmjt/2U08ZckEw4FU034wbWaeCaB5hq7mC6fNXl/
+      #            plaintext-password ""
+      #        }
+      #        full-name "John Smith"
+      #        level operator
+      #    }
+      #    user vyos {
+      #        authentication {
+      #            encrypted-password $1$5HsQse2v$VQLh5eeEp4ZzGmCG/PRBA1
+      #            plaintext-password ""
+      #        }
+      #        level admin
+      #    }
+      # }
+
+      # sometimes the hash is masked
+
+      # login {
+      #   user vyos {
+      #        authentication {
+      #            encrypted-password ****************
+      #            plaintext-password ""
+      #        }
+      #        level admin
+      #    }
+      # }
+
+      r =  'user ([^ ]+) {\s*authentication {\s*'
+      r << 'encrypted-password (\$?[\w$\./\*]*)\s*' # leading $ is optional incase the password is all stars
+      r << 'plaintext-password "([^"]*)"\s*'
+      r << '}(?:\s*'
+      r << 'full-name "([^"]*)")?\s*'
+      r << 'level (operator|admin)'
+      config.scan(/#{Regexp.new(r)}/mi).each do |result|
+        username = result[0].strip
+        hash = result[1].strip
+        plaintext = result[2].strip
+        # full-name is an optional field
+        # we label it, but dont actually use it.  Maybe future expansion?
+        unless result[3].nil?
+          name = result[3].strip
+        end
+        level = result[4].strip
+        cred = credential_data.dup
+        cred[:username] = username
+        unless hash.start_with?('********') # if not in config mode these are masked
+          cred[:jtr_format] = identify_hash(hash)
+          cred[:private_data] = hash
+          print_hash = " with hash #{hash}"
+        end
+        cred[:access_level] = level
+        create_credential_and_login(cred) if framework.db.active
+        unless plaintext.empty?
+          cred[:jtr_format] = ''
+          cred[:private_type] = :password
+          cred[:private_data] = plaintext
+          create_credential_and_login(cred) if framework.db.active
```

Comment:
I tend to save both, just in case.  To start, I wasn't able to actually get a `plaintext-password` (maybe that's an import only feature that gets blanked and hashed on import?).  I would also assume the hash matches the password, but often times thinking optimistically about code is a bad idea.  The other thing is, the password cracker modules defaults to using passwords in the DB, so this should be an easy/instant crack if done through MSF: https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/auxiliary/password_cracker.rb#L38